### PR TITLE
Add a user tutorial for configuring authorization for list-typed claims

### PR DESCRIPTION
**Pull request by lei-tang**
Add a user tutorial for configuring authorization for list-typed claims.

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -385,6 +385,8 @@ RPCs
 sa
 sayin
 schemas
+scope1
```
Comment (geeknoid) [inline .spelling]: I don't think scope1 and scope2 should be in the dictionary. Instead, put backticks around the use of these strings in the text.

Comment (rcaballeromx) [reply .spelling]: Even better, separate the term from the number. "scope 1" and "scope 2" cause no misspelling.

Comment (lei-tang) [reply .spelling]: Done: removed these two words from the dictionary.

**Comment by lei-tang** [comment]
Merged to https://github.com/istio/istio.io/pull/2652.

**Comment by lei-tang** [comment]
The merging to #2652 is per the requests from the reviewers' comments:
- Review comment 1: "@lei-tang @rcaballeromx @liminw
I suggest to merge to a single task. As the only difference between this PR and #2653 is the key name used in the request.auth.claims property.
This PR uses request.auth.claims[group] with JWT claim ["group1", "group2"] while #2653 uses request.auth.claims[scope] with JWT claim ["scope1", "scope2"].
They are testing the same functionality and code path and user is allowed to use any other key names, like my-custom-group or my-custom-scope if they want. You can mention this flexibility in the same task but I think there is no need to repeat it in a new task."
- Review comment 2: "@yangminzhu I fully agree. I think that just adding a pertinent section on the one document should be enough."

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+{{< text bash >}}
+$ cat <<EOF | kubectl apply -n $NS -f -
+apiVersion: "rbac.istio.io/v1alpha1"
+kind: ServiceRoleBinding
+metadata:
+  name: bind-httpbin-viewer
+  namespace: rbac-listclaim-test-ns
+spec:
+  subjects:
+  - user: "*"
```
Comment (liminw) [inline content/docs/tasks/security/rbac-list-claims/index.md]: Remove the line "user: "*"". 

```
  subjects:
  - properties:
       request.auth.claims[scope]: "scope1"
```

**Comment by liminw** [comment]
@lei-tang Can we add an example for "group" field in ServcieRoleBinding? We should use "groups" claim in this case.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
+---
+title: Authorization for list-typed claims
+description: Tutorial for configuring authorization for list-typed claims in Istio.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+weight: 10
+keywords: [security,authorization]
+---
+
+This task go through examples of authorization for list-typed claims in Istio.
+
+## Before you begin
+
+* Understand [authorization](/docs/concepts/security/#authorization) concepts and go through a guide for [configuring Istio authorization](/docs/tasks/security/role-based-access-control).
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+* Understand Istio [authentication policy](/docs/concepts/security/#authentication-policies) and related
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+[mutual TLS authentication](/docs/concepts/security/#mutual-tls-authentication) concepts.
+
+* Have a Kubernetes cluster with Istio installed, with mutual TLS enabled (e.g use `install/kubernetes/istio-demo-auth.yaml` as described in
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+[installation steps](/docs/setup/kubernetes/quick-start/#installation-steps)).
+
+### Setup
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+This tutorial runs in a new namespace called `rbac-listclaim-test-ns`, with two services, `httpbin` and `sleep`, both running with an Envoy sidecar proxy. The following commands set an environmental variable to store the name of the namespace, creates the namespace, and starts the two services. Before running the following commands, you need to enter the directory containing the Istio installation files.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+
+{{< text bash >}}
+$ export NS=rbac-listclaim-test-ns
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+$ kubectl create ns $NS
+$ kubectl apply -f <(istioctl kube-inject -f @samples/httpbin/httpbin.yaml@) -n $NS
+$ kubectl apply -f <(istioctl kube-inject -f @samples/sleep/sleep.yaml@) -n $NS
+{{< /text >}}
+
+To verify that `httpbin` and `sleep` services are running and `sleep` is able to reach `httpbin`, the following curl command should succeed with HTTP code 200.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+
+{{< text bash >}}
+$ kubectl exec $(kubectl get pod -l app=sleep -n $NS -o jsonpath={.items..metadata.name}) -c sleep -n $NS -- curl http://httpbin.$NS:8000/ip -s -o /dev/null -w "%{http_code}\n"
+200
+{{< /text >}}
+
+In addition, verify that no authentication policies have been defined in the newly created namespace.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+
+{{< text bash >}}
+$ kubectl get policies.authentication.istio.io -n $NS
+No resources found.
+{{< /text >}}
+
+## Configure authentication and authorization
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+
+### Configure JWT authentication with mutual TLS
+
+Apply an authentication policy to require both mutual TLS and JWT authentication for `httpbin`:
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+  - mtls: {}
+  origins:
+  - jwt:
+      issuer: "testing@secure.istio.io"
+      jwksUri: "{{< github_file >}}/security/tools/jwt/samples/jwks.json"
+  principalBinding: USE_ORIGIN
+EOF
+{{< /text >}}
+
+With the above authentication policy, a valid JWT is needed to access the httpbin service.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+The JWT must be signed by the JWKS endpoint defined in the above policy.
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+In this tutorial, the [JWKS endpoint]({{< github_file >}}/security/tools/jwt/samples/jwks.json) is
+from the Istio code base and the JWT is from a [sample JWT]({{< github_file >}}/security/tools/jwt/samples/groups-scope.jwt),
+which contains a JWT claim with the claim key being "scope" and the claim value being a list of
+strings [`"scope1"`, `"scope2"`]. To generate a JWT with other list-typed claims for testing purpose, you may use
+the `gen-jwt` [python script]({{<github_file>}}/security/tools/jwt/samples/gen-jwt.py).
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Changed to "You may use the gen-jwt python script to generate a JWT with other list-typed claims for testing purpose. Instructions for using gen-jwt.py are in the gen-jwt script."

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+Set the environmental variable TOKEN to contain a valid sample JWT token:
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+
+{{< text bash>}}
+$ TOKEN=$(curl {{< github_file >}}/security/tools/jwt/samples/groups-scope.jwt -s)
+{{< /text >}}
+
+With the authentication policy set, curl connection to the httpbin service should succeed if the JWT token is attached:
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+the curl connection to the httpbin service succeeds after the RBAC policy has become effective. The curl
+connection to the httpbin service succeeds because its header includes a valid JWT with the `scope` claim
+value [`"scope1"`, `"scope2"`] containing `scope1`.
+
+{{< text bash >}}
+$ kubectl exec $(kubectl get pod -l app=sleep -n $NS -o jsonpath={.items..metadata.name}) -c sleep -n $NS -- curl http://httpbin.$NS:8000/ip -s -o /dev/null -w "%{http_code}\n" --header "Authorization: Bearer $TOKEN"
+200
+{{< /text >}}
+
+## Cleanup
```
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

File: content/docs/tasks/security/rbac-list-claims/index.md (hunk 2 above)
Comment (lei-tang) [reply content/docs/tasks/security/rbac-list-claims/index.md]: Done.

**Comment by lei-tang** [comment]
@liminw This PR has been merged to #2652, which contains an example for  "groups" claim.

**Pull request closed without merging**